lib/cuda/cudaDeviceInterface.cpp

```diff
@@ -24,6 +44,47 @@ cudaDeviceInterface::~cudaDeviceInterface() {
     cleanup_memory();
 }
 
+void cudaDeviceInterface::set_device_clocks(unsigned int mem_clock, unsigned int core_clock,
+                                            unsigned int mem_count, unsigned int core_count) {
+
+    // Set default clocks to zero
+    uint32_t gpu_mem_clock = config.get_default<uint32_t>(unique_name, "gpu_mem_clock", 0);
+    uint32_t gpu_core_clock = config.get_default<uint32_t>(unique_name, "gpu_core_clock", 0);
+
+    uint32_t get_gpu_mem_clock, get_gpu_core_clock;
+
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_mem_clock, get_gpu_mem_clock);
+    nvmlDeviceGetMaxClockInfo (gpu_id_, gpu_core_clock, get_gpu_core_clock);
+
+    if (get_gpu_mem_clock != 0 & get_gpu_core_clock != 0) {
+
+        // Get and update the GPU clocks from the config file
+        gpu_mem_clock = config.get<uint32_t>(unique_name, "gpu_mem_clock");
+        gpu_core_clock = config.get<uint32_t>(unique_name, "gpu_core_clock");
+
+        for (int i = 0; i < mem_count; ++i)  {
+             mem_clock[i] == get_gpu_mem_clock;
+             break;
+        }
+
+        if (i == mem_count)
+            gpu_mem_clock = config.get_default<uint32_t>(unique_name, "gpu_mem_clock", 0)
```

Comment:
This function won't return anything different if the clocks don't match one of the clocks in the list.

The optimal thing to do here is find the closest clock (rounding up) from the supported clocks list and use that one.   e.g. if the allowed clocks are 900, 1000, and 1200 and someone sets the clock to 1100, then round up to 1200.  Or if the user sets 1300, then round down to the highest clock option - again 1200 in this example.

Reply:
Added a patch for finding the clock closest to the list and setting the clock. The comparison is made on the integer values of the frequency clocks.